website/docs/r/ovh_domain_zone.html.markdown

```diff
@@ -57,9 +57,9 @@ The following arguments are supported:
   * `plan_code` - (Required) Plan code
   * `pricing_mode` - (Required) Pricing model identifier
   * `catalog_name` - Catalog name
-  * `configuration` - (Optional) Representation of a configuration item for personalizing product
-    * `label` - (Required) Identifier of the resource
-    * `value` - (Required) Path to the resource in API.OVH.COM
+  * `configuration` - (Required) Representation of a configuration item for personalizing product. 2 configurations are required : one for `zone` and one for `template`
+    * `label` - (Required) Identifier of the resource : `zone` or `template`
+    * `value` - (Required) For `zone`, the value is the zone name `myzone.example.com`. For `template`, the value can be `basic`, `minimized` or  `redirect` which is the same as `minimized` with additional entries for a redirect configuration.
```

Comment:
@rbeuque74

Which DNS records do I get on each of the `template` options (`basic`, `minimized`, or  `redirect`)?.

A little table (or even better an example zone file) could help to visualize what you get on each template option.

See this example from #505, this is what I got by using the `minimized` _template_:

> ```
> $TTL 3600
> @       IN SOA dns10.ovh.ca. tech.ovh.net. (2023113003 86400 3600 3600000 60)
>            IN NS     dns10.ovh.ca.
>            IN NS     ns10.ovh.ca.
>            IN A      144.217.153.176
>            IN TXT    "1|www.terraform.example.com"
>        600 IN TXT    "v=spf1 include:mx.ovh.ca ~all"
> www        IN A      144.217.153.176
> www        IN TXT    "3|welcome"
> ```

At the end of the day DNS is one crucial part of the infrastructure that is often overlooked.
Terraform provides a way to declare the exact resources that you want, nothing more, nothing less.

| _It's **always** DNS_ |
|:----:|
| ![][its-always-dns]

I'd even advise for this resource to create an "empty" zone with the bare minimum resources (`SOA` and `NS`) just like the [`aws route53_zone`][aws-route53-zone] resource does.
Then use a few [`ovh_domain_zone_record`][ovh_domain_zone_record] resources to specify exactly which DNS records you want.

You could even put the `basic`, `minimized`, and `redirect` _templates_ in the docs, so we can copy-paste and adjust as needed to have a well defined infrastructure.

[its-always-dns]: https://dnx.solutions/wp-content/uploads/2021/02/Its-always-dns.jpeg

[aws-route53-zone]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone

[ovh_domain_zone_record]: https://registry.terraform.io/providers/ovh/ovh/latest/docs/resources/ovh_domain_zone_record

--------

Related to: #505